corehq/apps/locations/util.py

```diff
@@ -321,11 +321,3 @@ def get_locations_from_ids(location_ids, domain, base_queryset=None):
     if len(locations) != expected_count:
         raise SQLLocation.DoesNotExist('One or more of the locations was not found.')
     return locations
-
-
-def get_lineage_from_location_id(location_id):
-    return get_lineage_from_location(Location.get(location_id))
-
-
-def get_lineage_from_location(location):
-    return list(reversed(location.path))
```

Comment:
I'm not sure exactly how the original problem manifested, but this could've been a source - `path` includes `self`, but `lineage` isn't supposed to.  Additionally, `path` was defined as `lineage` + `self._id`, so I could see those working together to repeatedly add on ids.